powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/enums/PowerAuthSignatureTypes.java

```diff
@@ -77,12 +77,14 @@
      * @return Enum value.
      */
     public static PowerAuthSignatureTypes getEnumFromString(String value) {
-        PowerAuthSignatureTypes type = map.get(value.toLowerCase());
-        if (type == null) { // try to guess the most usual suspect...
-            return PowerAuthSignatureTypes.POSSESSION_KNOWLEDGE;
-        } else {
-            return type;
+        if (value != null) {
+            final PowerAuthSignatureTypes type = map.get(value.toLowerCase());
+            if (type != null) {
+                return type;
+            }
         }
+        // Otherwise try to guess the most usual suspect...
+        return PowerAuthSignatureTypes.POSSESSION_KNOWLEDGE;
```

Comment:
What if some uses a signature type with a typo, it will fall back to `POSSESSION_KNOWLEDGE` and then work magically? Shouldn't we rather prevent this case?